dashboard/app/helpers/proxy_helper.rb

```diff
@@ -0,0 +1,70 @@
+require 'net/http'
+require 'uri'
+
+module ProxyHelper
+  include DomainHelper
+
+  def render_proxied_url(location, allowed_content_types:, expiry_time:, infer_content_type:, redirect_limit: 5)
```

Comment:
Could you please comment the enforcement of valid top-level domains? Many readers might  miss this subtlety in the code below.